src/map.ts

```diff
@@ -0,0 +1,2 @@
+const myMap = Map;
```

Comment:
This looks strange, can you explain why this is needed? At very least it looks like we probably don't need a whole file for this

Reply:
So I believed this was getting polyfilled by babel this was my way of plucking out the Map constructor and the generated code here would look like an import or injection of the Map type. Turns out that's not the case so I'm back to manually importing it. Which... breaks the bundle again, not sure why but commonjs isn't translating the core-js dependency. I'm ready to let this goal go.

In the latest commit I brought back the old polyfil, its not used by most but I can improve it in the next cycle of typing, if that makes sense?